crates/core/src/writer/stats.rs

```diff
@@ -231,18 +267,8 @@ impl StatsScalar {
                     v.max_bytes()
                 };
 
-                let val = if val.len() <= 4 {
-                    let mut bytes = [0; 4];
-                    bytes[..val.len()].copy_from_slice(val);
-                    i32::from_be_bytes(bytes) as f64
-                } else if val.len() <= 8 {
-                    let mut bytes = [0; 8];
-                    bytes[..val.len()].copy_from_slice(val);
-                    i64::from_be_bytes(bytes) as f64
-                } else if val.len() <= 16 {
-                    let mut bytes = [0; 16];
-                    bytes[..val.len()].copy_from_slice(val);
-                    i128::from_be_bytes(bytes) as f64
+                let val = if val.len() <= 16 {
+                    i128::from_be_bytes(sign_extend_be(val)) as f64
```

Comment:
Looks like this was causing those [test failures](https://github.com/delta-io/delta-rs/actions/runs/9004694154/job/24740575061?pr=2491#step:9:680). 

In particular note that the above would wrongly extend slices that are not a power of 2 long. This is the case that occured for the failing tests, since this:
https://github.com/delta-io/delta-rs/blob/81593e919497221a1a08bf8db9d20e8e4a39a8a6/python/tests/conftest.py#L218
would result in min [0, 39, 16] and max [0, 54, 176] fized length byte arrays, and these would in turn be coerced to [0, 39, 16, 0] and max [0, 54, 176, 0], instead of [0, 0, 39, 16] and max [0, 0, 54, 176].

Contrast with arrow-rs/datafusion where the extension happens at the beginning of the array:
https://github.com/apache/arrow-rs/blob/b25c441745602c9967b1e3cc4a28bc469cfb1311/parquet/src/arrow/array_reader/fixed_len_byte_array.rs#L170

Granted I'm not sure what/how encodes the min value "10.000" to [0, 39, 16] in the first place, aside from that it occurs in pyarrow.

Reply:
Hi @ion-elgreco, can you approve the rest of the workflows?